### PR TITLE
add and format the version fallback APIs

### PR DESCRIPTION
**Pull request by tigershi**

### CODE REVIEW

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationWithPatternAction.java [hunk 1]
```diff
@@ -52,12 +52,20 @@
 	   public APIResponseDTO getTransPattern(TranslationWithPatternDTO data) throws Exception {
 	      logger.info("begin getTransPattern");
 	      if (validate(data)) {
+	    	  List<String> availableVersions = null;
+	          try {
+	        	  availableVersions = productService.getSupportVersionList(data.getProductName());
+	          }catch(L3APIException e) {
+	        	  availableVersions = null;
```
Comment (linr211) [inline g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationWithPatternAction.java]: availableVersions is null if exception happen here, so this line is extra and please remove it.

Comment (tigershi) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationWithPatternAction.java]: done

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java [hunk 2]
```diff
@@ -211,14 +212,17 @@ public APIResponseDTO getPartialComTrans(String productName,
...
                   ja.add(jsonNullObj);
               }
                resulttranslationDTO.setBundles(ja);
-              return  super.handleResponse(APIResponseStatus.MULTTRANSLATION_PART_CONTENT, resulttranslationDTO);
-              
+          	 if(oldVersion.equals(version)) {
+    			 return handleResponse(APIResponseStatus.MULTTRANSLATION_PART_CONTENT, resulttranslationDTO);
+    		 }else {
+    			 return handleResponse(APIResponseStatus.VERSION_FALLBACK_TRANSLATION, resulttranslationDTO);
+    		 }
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java]: Please use handleVersionFallbackResponse

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/BaseAction.java [hunk 3]
```diff
@@ -8,13 +8,44 @@
...
+        }catch(Exception e) {
+          return version;
+        }
+    }
+   
+	/**
+	 * This method use to handle and package the version fallback response content 
+	 * 
+	 */
+	protected APIResponseDTO handleVersionFallbackResponse(String oldVersion, String newVersion, Object data) {
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/BaseAction.java]: How about changing name to handleTranslationResponse?

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java (hunk 2 above)
Comment (tigershi) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java]: this method return APIResponseStatus.MULTTRANSLATION_PART_CONTENT , not APIResponseStatus.OK. so can't use handleVersionFallbackResponse

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/BaseAction.java (hunk 3 above)
Comment (tigershi) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/BaseAction.java]: the handleVersionFallbackResponse method not only use translation response but also use the get support language API and get component API, So the name changes to handleTranslationResponse is inappropriate.

**Pull request merged**